rfc/0011-audit-checks.md

```diff
@@ -0,0 +1,354 @@
+
+Targeting all kind of resources would make more complicated to find all the
+Kubernetes resources contained inside of a Namespace that have to be audited.
+
+To simplify things, we can state that policies targeting `*` are not going to be
+evaluated by the audit checks. Users have to be explicit about the
+Kubernetes resources a policy targets.
+
+### No live update of `PolicyResult` objects
+
+The proposed workflow requires that only one process is actively auditing the
+resources of a given Namespace.
+
+That's because:
+
+* The `PolicyResult` of the Namespace is used to reduce the
+  number of auditing evaluations performed
+* Once all the policies and involved resources are audited, the
+  `PolicyReport` audit is going to be overwritten with the results of the audit
+
+Users can still get live data about the rejections/approvals by looking both
+at Prometheus data and Open Telemetry traces.
+
+We should focus on making the process of creating a `PolicyReport` as fast as
+possible. The reports are recreated on a regular basis. If a user wants a fresh
+`PolicyReport` without having to wait for the next scheduled run, we can
+provide a way for the user to trigger this operation.
+
+Finally, if we really have to implement live update of `PolicyReport`, we can
+come up with an alternative architecture that will allow for that. For example,
+we could have all the write operations of `PolicyReport` to go through a
+dedicated "writer" service.
+
+## Creating a `PolicyReport` object
+
+The following section describes how, given a Namespace, we will produce a `PolicyReport` describing the compliance status.
+
+The process to create the `PolicyReport` for a single Namespace can be applied also to generate the `ClusterPolicyReport`, which is the one targeting only cluster-wide resources.
+
+### Step 1: create association between `ClusterAdmissionPolicy` and namespaces
+
+`ClusterAdmissionPolicy` by default inspect events happening inside of all the namespaces, however users can limit their scope
+by using the `spec.namespaceSelector` field.
+
+We need to create a dictionary that holds the following information:
+
+  * key: name of the `ClusterAdmissionPolicy`
+  * value: a list containing the names of the namespaces that are relevant to the `ClusterAdmissionPolicy`
+
+This operation can be done by iterating over each `ClusterAdmissionPolicy` and
+performing a "list namespaces" query against the API server that uses the
+same filter specified by the `ClusterAdmissionPolicy`.
+
+### Step 2: find relevant policies
+
+Given a Namespace to be audited, create a list of all the policies that are interested about the Namespace.
+
+All the `AdmissionPolicy` resources that are defined inside of the namespace are part of the list.
+
+For `ClusterAdmissionPolicy` resources, use the information stored inside of the
+dictionary that was created during the previous step.
+
+> Note: we will consider only policies that inspect `CREATE` events
```

Comment:
I would put this as part of the `### No audit for policies targeting `*`` paragraph. Listing the precondition of policies to be audited all in the same paragraph.